packages/grafana-runtime/src/services/dashboard-srv/DashboardSrvSingleton.ts

```diff
@@ -0,0 +1,98 @@
+import { PluginsAPIPanelModelWrapper } from './PanelWrapper';
+import {
+  CoreDashboardModel,
+  CoreDashboardSrv,
+  CorePanelModel,
+  PluginsAPIDashboardModel,
+  PluginsAPIDashboardSrv,
+  PluginsAPIPanelModel,
+} from './types';
+
+export class PluginsAPIDashboardModelWrapper implements PluginsAPIDashboardModel {
+  #dashboard: CoreDashboardModel | undefined;
+
```

Comment:
I think the wrapper implementations can live in core, only the interfaces need to be in grafana-runtime  (then there is no need for the types CoreDashboardModel type)

Reply:
I like the idewa. It is a good solution to the whole interfaces and types problem that makes me write all these middle-way interfaces. Another solution I considered was abstracting `DashboardSrv` out to grafana/runtime but that would had taken considerably more work.

Reply:
Resolution: moved the wrapper implementation inside grafana core.